code/clue.js

```diff
@@ -48,12 +278,59 @@ const pickKiller = () => {
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = ` Age:${mystery.killer.age}`
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+  let favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favouriteWeapon)
+  document.getElementById('killerFavouriteWeapon').innerHTML = `Favourite weapon: ${favWeapon.name}`
+  document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`
+}
+document.getElementById('killerCard').onclick = pickKiller
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
```

Comment:
You can remove the extra lines here and there and your code will be perfect! (at least from what I can tell 😃)